src/Lykke.Service.Stellar.Api.Services/Transaction/TransactionService.cs

```diff
@@ -210,36 +212,37 @@ private static string GetErrorMessage(Exception ex)
         {
             var errorMessage = ex.Message;
             // ReSharper disable once InvertIf
-            if (ex is HttpResponseException badRequest)
+            if (ex is BadRequestHorizonApiException badRequest)
             {
                 //TODO:
-                //var resultCodes = JsonConvert.SerializeObject(badRequest.ErrorDetails.Extras.ResultCodes);
-                //errorMessage += $". ResultCodes={resultCodes}";
+                var resultCodes = JsonConvert.SerializeObject(badRequest.ErrorCodes);
+                errorMessage += $"{badRequest.Message}. ResultCodes={resultCodes}";
             }
+
             return errorMessage;
         }
 
         private static TxExecutionError GetErrorCode(Exception ex)
         {
             //TODO:
-            //var bre = ex as HttpResponseException;
-            //var resultCodes = bre?.ErrorDetails?.Extras?.ResultCodes;
-            //var ops = resultCodes?.Operations;
-            //var transactionDetail = resultCodes?.Transaction;
-            //if (bre?.ErrorDetails != null && bre.ErrorDetails.Status == (int)HttpStatusCode.BadRequest
-            //    && ops != null
-            //    && ops.Length > 0
-            //    && (ops[0].Equals(StellarSdkConstants.OperationUnderfunded)
-            //        || ops[0].Equals(StellarSdkConstants.OperationLowReserve)))
-            //{
-            //    return TxExecutionError.NotEnoughBalance;
-            //}
-
-            //if (transactionDetail == "tx_too_late" ||
-            //    transactionDetail == "tx_bad_seq")
-            //{
-            //    return TxExecutionError.BuildingShouldBeRepeated;
-            //}
+            var bre = ex as BadRequestHorizonApiException;
+            var resultCodes = bre?.ErrorCodes;
+            var transactionDetail = bre?.Message;
+
+            if (transactionDetail != null
+                && resultCodes != null
+                && resultCodes.Length > 0
+                && (resultCodes[0].Equals(StellarSdkConstants.OperationUnderfunded)
```

Comment:
Why do we check exactly first item, isn't `Contains` fee better here?

Reply:
nope, It is how it worked before, we do not need to fix this